rules/prefer-negative-index.js

```diff
@@ -0,0 +1,303 @@
+				'Array',
+				'String',
+				'ArrayBuffer',
+				'Int8Array',
+				'Uint8Array',
+				'Uint8ClampedArray',
+				'Int16Array',
+				'Uint16Array',
+				'Int32Array',
+				'Uint32Array',
+				'Float32Array',
+				'Float64Array',
+				'BigInt64Array',
+				'BigUint64Array'
+				// 'Blob'
+				// 'File'
+			])
+		}
+	],
+	[
+		'splice',
+		{
+			argumentsIndexes: [0],
+			supportObjects: new Set([
+				'Array'
+			])
+		}
+	]
+]);
+
+const OPERATOR_MINUS = '-';
+
+const isPropertiesEqual = (node1, node2) => properties => {
+	return properties.every(property => isEqual(node1[property], node2[property]));
+};
+
+const isTemplateElementEqual = (node1, node2) => {
+	return node1.value &&
+		node2.value &&
+		node1.tail === node2.tail &&
+		isPropertiesEqual(node1.value, node2.value)(['cooked', 'raw']);
+};
+
+const isTemplateLiteralEqual = (node1, node2) => {
+	const {quasis: quasis1} = node1;
+	const {quasis: quasis2} = node2;
+
+	return quasis1.length === quasis2.length &&
+		quasis1.every((templateElement, index) => isEqual(templateElement, quasis2[index]));
+};
+
+const isEqual = (node1, node2) => {
+	if (node1 === node2) {
+		return true;
+	}
+
+	const compare = isPropertiesEqual(node1, node2);
+
+	if (!compare(['type'])) {
+		return false;
+	}
+
+	const {type} = node1;
+
+	switch (type) {
+		case 'Identifier':
+			return compare(['name', 'computed']);
+		case 'Literal':
+			return compare(['value', 'raw']);
+		case 'TemplateLiteral':
+			return isTemplateLiteralEqual(node1, node2);
+		case 'TemplateElement':
+			return isTemplateElementEqual(node1, node2);
+		case 'BinaryExpression':
+			return compare(['operator', 'left', 'right']);
+		case 'MemberExpression':
+			return compare(['object', 'property']);
+		default:
+			return false;
+	}
+};
+
+const isLengthMemberExpression = node => node &&
+	node.type === 'MemberExpression' &&
+	node.property &&
+	node.property.type === 'Identifier' &&
+	node.property.name === 'length' &&
+	node.object;
+
+const isLiteralPositiveValue = node =>
+	node &&
+	node.type === 'Literal' &&
+	typeof node.value === 'number' &&
+	node.value > 0;
+
+const getLengthMemberExpression = node => {
+	if (!node) {
+		return;
+	}
+
+	const {type, operator, left, right} = node;
+
+	if (
+		type !== 'BinaryExpression' ||
+		operator !== OPERATOR_MINUS ||
+		!left ||
+		!isLiteralPositiveValue(right)
+	) {
+		return;
+	}
+
+	if (isLengthMemberExpression(left)) {
+		return left;
+	}
+
+	// Nested BinaryExpression
+	return getLengthMemberExpression(left);
+};
+
+const getRemoveAbleNode = (target, argument) => {
+	const lengthMemberExpression = getLengthMemberExpression(argument);
+
+	if (lengthMemberExpression && isEqual(target, lengthMemberExpression.object)) {
+		return lengthMemberExpression;
+	}
+};
+
+const getRemovalRange = (node, sourceCode) => {
+	let before = sourceCode.getTokenBefore(node);
+	let after = sourceCode.getTokenAfter(node);
+
+	let start = node.range[0];
+	let end = node.range[1];
```

Comment:
```suggestion
	let [start, end] = node.range;
```

Same with the below.

Reply:
I forgot why I seperate them... maybe mistake

Reply:
assign can't do this, different ranges

Reply:
```js
let [start, end] = [1, 2];
console.log(start, end);
//=> 1  2
```

Reply:
I know this, already did